controllers/clusters/on_premises.go

```diff
@@ -925,3 +974,104 @@ func newClusterIPService(
 		},
 	}
 }
+
+func getReplacedNodes(oldNodesDC, newNodesDC *v1beta1.DataCentreStatus) (newNode, oldNode *v1beta1.Node) {
+	for _, aNode := range oldNodesDC.Nodes {
+		for _, bNode := range newNodesDC.Nodes {
+			if aNode.ID == bNode.ID {
+				oldNode = nil
+				newNode = nil
+				break
+			}
+			oldNode = aNode
+			newNode = bNode
+		}
+
+		if oldNode != nil {
+			return
+		}
+	}
+
+	return nil, nil
+}
+
+func deleteReplacedNodeResources(ctx context.Context, k8sClient client.Client, oldID string) (deleteUsed bool, err error) {
+	vmList := &virtcorev1.VirtualMachineList{}
+	err = k8sClient.List(ctx, vmList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			models.NodeIDLabel: oldID,
+		}),
+	})
+	if err != nil {
+		return false, err
+	}
+
+	for _, vm := range vmList.Items {
+		deleteUsed = true
+		err = k8sClient.Delete(ctx, &vm)
+		if err != nil {
+			return false, err
+		}
+
+		patch := client.MergeFrom(vm.DeepCopy())
+		controllerutil.RemoveFinalizer(&vm, models.DeletionFinalizer)
+		err = k8sClient.Patch(ctx, &vm, patch)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	dvList := &cdiv1beta1.DataVolumeList{}
+	err = k8sClient.List(context.TODO(), dvList, &client.ListOptions{
```

Comment:
Please change `context.TODO()` to `ctx` from functions arguments and change for all such cases

Reply:
Done